src/ensembl/src/shared/components/checkbox/Checkbox.tsx

```diff
@@ -55,7 +55,7 @@ const Checkbox = (props: CheckboxProps) => {
     props.onChange(!props.checked);
   };
 
-  const styles = { ...defaultStyles };
+  const styles = props.classNames ? { ...defaultStyles } : defaultStyles; // this fix the test cases and also if there is no classNames passed as props
```

Comment:
The reason the tests failed, as you might have learned from @imransl,  is because Jest has been set up to use `identity-object-proxy` when it's importing `css` or `scss` files (https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/jest.config.js#L10). I don't know if that was a good idea, but as a result the `defaultStyles` that gets imported in the test is an ES6 Proxy rather than a good old plain object. And ES6 Proxies cannot be iterated using a `for ... in` loop.

The code above is just a duct tape around the problem; the actual problem still remains, but we just happen not to have a test revealing it. I suspect that if you had a test similar to the [one we are running for Checkbox label](https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/shared/components/checkbox/Checkbox.test.tsx#L46-L53), but that would test the classes of the checkbox element instead, this test would fail.

Reply:
@azangru your commit is good and make the code more understandable, thanks. Just one thing i wanted to confirm is the class checkboxHolder in the component is something that cannot be overwritten from props, is that the correct behavior we want/expect.

Reply:
Thanks, @ens-ma7, that's my mistake. I'll fix that and add a couple of tests to make sure that classes are applied correctly.